### ISSUE TITLE
[okta-angular] move router navigation from service to callback.component 

### ISSUE THREAD

**Issue by ruhkopf** [okta/okta-oidc-js#441]
## I'm submitting this issue for the package(s):

- [ ] jwt-verifier
- [x] okta-angular
- [ ] oidc-middleware
- [ ] okta-react
- [ ] okta-react-native
- [ ] okta-vue

## I'm submitting a:

- [ ] Bug report  
- [x] Feature request
- [ ] Other (Describe below)

## Current behavior
We have some custom logic that we need to do on each login. (I'd say, this is a pretty common requirement.) At the moment I can't easily do this, because the [Callback Component](https://github.com/okta/okta-oidc-js/blob/master/packages/okta-angular/src/okta/components/callback.component.ts) delegates to the service and the service [instructs the router to navigate away](https://github.com/okta/okta-oidc-js/blob/master/packages/okta-angular/src/okta/services/okta.service.ts#L202).

## Expected behavior
It would be much nicer if the service would just handle the business logic of handling the token etc, but the component would actually do the routing. This way I could easily implement my own component doing the logic I need to do, for example:

```ts
export class MyAuthCallbackComponent implements OnInit {
  constructor(private oktaAuthService: OktaAuthService) {}

  async ngOnInit() {
    await this.handleAuthentication()
    // do my own logic here
    const fromUri = this.oktaAuthService.getFromUri()
    this.router.navigate([fromUri.uri], fromUri.extras)
  }
}

```

## Extra information about the use case/user story you are trying to implement
Basically I need to perform business logic whenever the user logs in. I don't want to do this on every page refresh or anything, just when the user logs in.

**Comment by swiftone** [comment]
INFO: This relates to the discussion about organizing routing

**Comment by swiftone** [comment]
@ruhkopf - As a workaround for your need, you should be able to call the parts from INSIDE handleAuthentication itself in your MyAuthCallbackComponent.  (https://github.com/okta/okta-oidc-js/blob/master/packages/okta-angular/src/okta/services/okta.service.ts#L185).   Just don't call the navigate() part.

If this doesn't work for you, please let us know.

**Comment by ruhkopf** [comment]
Yep, that's what I'm currently doing and it works. However, this means accessing private fields. Would you consider declaring `oktaAuth` and `emitAuthenticationState` public or refactor the code as suggested in this issue?

**Comment by aarongranick-okta** [comment]
@ruhkopf Thank you for raising this issue! We have discussed this issue internally and are considering the options. There is no danger in accessing those "private" fields. If that is working for you, we will consider this method a temporary workaround while we develop a better fix. Thanks again!

**Comment by ruhkopf** [comment]
Sounds good, thank you.
